src/main/java/com/microsoft/serviceClient/GraphServiceClient.java

```diff
@@ -0,0 +1,79 @@
+package com.microsoft.serviceClient;
+
+import com.azure.core.credential.TokenCredential;
+import com.microsoft.graph.Requests.BaseGraphRequestAdapter;
+import com.microsoft.graph.Requests.IBaseClient;
+import com.microsoft.graph.httpcore.middlewareoption.GraphClientOptions;
+import com.microsoft.info.Constants;
+import com.microsoft.kiota.RequestAdapter;
+import com.microsoft.kiota.authentication.AnonymousAuthenticationProvider;
+import com.microsoft.kiota.authentication.AuthenticationProvider;
+import com.microsoft.kiota.authentication.AzureIdentityAuthenticationProvider;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import okhttp3.OkHttpClient;
+
+import javax.annotation.Nonnull;
+
+public class GraphServiceClient extends microsoft.graph.BaseGraphServiceClient implements IBaseClient {
+
+    private RequestAdapter requestAdapter;
```

Comment:
I don't think the field should be defined again here, we should refer directly to the field in the class that's generated.

Reply:
Unfortunately that would mean that we could not change the RequestAdapter since we do not have get/set methods in the generated class, hence the need for this field here. Unless the idea is that we only change the RequestAdapter when we create a new GraphServiceClient using 'new GraphServiceClient(requestAdapter)' which would require a new instantiation of the GraphServiceClient. An alternative would be to simply add the get/set methods to the generated class by implementing IBaseClient in BaseGraphServiceClient and thus remove the need for GraphServiceClient to implement IBaseClient and instead only require the extension of BaseGraphServiceClient.

Reply:
right because it's private for now. And @andrueastman had the [same problem with dotnet](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/a6059f734d959a454e5017a0372d5feacbc238dd/src/Microsoft.Graph/GraphServiceClient.cs#L73)
What would you think of a common refiner in Kiota to switch this property to public just for the API client class (not the other request builders) or add accessors?

Reply:
I think accessors would be a better method, it would make it obvious that we intend to grant the user the ability to get/set them in the client class. Do we want to add the implements IBaseClient to the api client class? Or would adding that mean having a direct dependency on core which would unnecessary for other generated API's?

Reply:
You can also add the accessors to the IBaseClient if you need to access the request adapter in core, which you'll need for large file upload and batch support.
As long as they match what we generate, the resulting api client will satisfy the interface. They won't need to be defined in this hand crafted client anymore however since they'll come from the generated class.

Reply:
https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/v3/telemetryRefactor/src/main/java/com/microsoft/graph/Requests/IBaseClient.java
The IBaseClient really only contains the accessors, I say we just add them to the generated class/implement the interface and we should be good to go

Reply:
I'll delete the accessors and the RequestAdapter property from this GraphServiceClient class